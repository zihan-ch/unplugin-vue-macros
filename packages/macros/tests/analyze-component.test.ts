import { initContext } from '@vue-macros/common'
import { describe, expect, test } from 'vitest'

describe('analyzeComponent', () => {
  describe('props', () => {
    test('object props', () => {
      const { ctx } = initContext(
        `<script setup>
  defineProps(props)
  </script>`,
        'basic.vue'
      )
      expect(ctx.component).toMatchSnapshot()
    })

    test('typescript props', () => {
      const { ctx } = initContext(
        `<script setup lang="ts">
  defineProps<{ title: string }>()
  </script>`,
        'basic.vue'
      )
      expect(ctx.component).toMatchSnapshot()
    })
  })
})
