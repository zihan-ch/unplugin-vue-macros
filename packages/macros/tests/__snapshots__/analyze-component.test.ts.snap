// Vitest Snapshot v1

exports[`analyzeComponent > props > object props 1`] = `
{
  "props": {
    "code": "props",
    "type": "unknown",
  },
}
`;

exports[`analyzeComponent > props > typescript props 1`] = `
{
  "props": undefined,
}
`;

exports[`analyzeComponent > props 1`] = `
{
  "props": {
    "code": "props",
    "type": "unknown",
  },
}
`;
